lib/__tests__/ModuleFinder-test.js

```diff
@@ -0,0 +1,94 @@
+import path from 'path';
+
+import ModuleFinder from '../ModuleFinder';
+import findPackageDependencies from '../findPackageDependencies';
+import normalizePath from '../normalizePath';
+import readFile from '../readFile';
+import requireResolve from '../requireResolve';
+
+jest.mock('../readFile');
+jest.mock('../findPackageDependencies');
+jest.mock('../requireResolve');
+
+let moduleFinder;
+
+function addFiles(files) {
+  return moduleFinder.handleFilesAdded(files.map((file) => ({
+    path: normalizePath(file, process.cwd()),
+    mtime: Date.now(),
+  })));
+}
+
+beforeEach(() => {
+  readFile.mockImplementation(() => Promise.resolve('export default {}'));
+
+  findPackageDependencies.mockImplementation(() => ['pacman', 'react-packer']);
+
+  requireResolve.__addResolvedPath(
+    path.join(process.cwd(), 'node_modules/pacman'),
+    path.join(process.cwd(), 'node_modules/pacman/index.js'),
+  );
+
+  requireResolve.__addResolvedPath(
+    path.join(process.cwd(), 'node_modules/react-packer'),
+    path.join(process.cwd(), 'node_modules/react-packer/index.js'),
+  );
+
+  moduleFinder = ModuleFinder.getForWorkingDirectory(process.cwd(), {
+    excludes: [],
+    ignorePackagePrefixes: ['react-'],
+  });
+  return moduleFinder.initializeStorage(':memory:');
```

Comment:
I'm not the biggest fan of having a bunch of things happen in beforeEach, because it makes each individual test more difficult to reason about and it can make it more difficult to add new tests to the file. I think this test is maybe small enough and the setup is maybe large enough so that it is okay, but it might be something worth considering inlining into each test for maximum clarity.